### PR TITLE
Improved attachments

### PR DESCRIPTION
**Pull request by sebastienros**

### CODE REVIEW

**Comment by sebastienros** [comment]
Note you can't test it until the servers are up to date as there is a new state for the server job.

File: src/BenchmarksDriver/Program.cs [hunk 1]
```diff
@@ -1231,6 +1245,44 @@ public static int Main(string[] args)
...
+                    return 8;
+                }
+
+                var destinationFilename = outputFileSegments.Length > 1
+                    ? outputFileSegments[1]
+                    : Path.GetFileName(attachmentFilename);
+
+                var requestContent = new MultipartFormDataContent();
+
+                var fileContent = attachmentFilename.StartsWith("http", StringComparison.OrdinalIgnoreCase)
```
Comment (davidfowl) [inline src/BenchmarksDriver/Program.cs]: var fileContent = attachmentFilename.StartsWith("http", StringComparison.OrdinalIgnoreCase)
                    ? new StreamContent(await _httpClient.GetByteArrayAsync(attachmentFilename))
                    : new StreamContent(new FileStream(attachmentFilename,FileMode.Open,FileAccess.Read,FileShare.ReadWrite,1,FileOptions.Asynchronous | FileOptions.SequentialScan));

Comment (sebastienros) [reply src/BenchmarksDriver/Program.cs]: Is `1` for the buffer size intentional? default is 4K

Comment (benaadams) [reply src/BenchmarksDriver/Program.cs]: `0` throws, `1` says don't use internal buffer

Comment (benaadams) [reply src/BenchmarksDriver/Program.cs]: Came up in https://github.com/aspnet/KestrelHttpServer/pull/2049#discussion_r165830218

image

**Pull request merged**